mantidproject/mantid
# IN16B properly modelled

**Description of work.**
This replaces the existing definition of IN16B with a proper model, that has physical and neutronic positions defined. 
This will have a retroactive effect, however the existing tests should still pass because the detectors are grouped tube-by-tube and hence the off-plane effects are not captured.

**To test:**

LoadEmptyInstrument of IN16B. 

With instrument view:
Should have one monitor, 8 single detectors in the scattering plane, and 16 PSD tubes vertically placed.

With show detectors:
Monitor should have ID=0, pixels in the PSD tubes: 1-2048, single detectors: 2049-2056.

**Merge also PR [#118](https://github.com/mantidproject/mantidgeometry/pull/118) in mantidgeometry.** 

To check the neutronic positions, run this in mantidgeometry:
`python ILL/IDF/in16b_bats_generateIDF.py --geometrytype N`

This will generate the neutronic positions as physical, so the generated IDF can be loaded and viewed in the instrument view.

In this view you should see stretched banana shapes for the reflections of the straight vertical tubes in the spherical mirror of the analysers focused on sample.



Part of #25575 . 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
